### DISCUSSION TITLE
20220703

### DISCUSSION THREAD

**Discussion by tmiw** [ARSFI/paclink-open-source#30] [General]
This is a pre-production build intended to gauge current status of the project and determine what needs to be fixed. Brief testing indicates that packet (using Kenwood TM-D700A), AGW (using AGWPE and UZ7HO soundmodem) and Telnet channels work; Pactor is TBD.

Contains (vs. [last pre-release version](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220617)):

* Fix bug preventing clearing of old channel list during channel list updates.
* PR #29: remove unused NLog.Targets.Gelf dependency.

Instructions:

1. Unzip paclink-open-source.zip somewhere on your hard drive.
2. Run setup.exe to install on your system. Paclink will be installed in the Start menu.
3. Open Paclink from the Start menu and reconfigure if required. (There is no import functionality from the existing Paclink at this time but if you used a pre-release version, configuration should be saved.)

Please leave comments in the release's discussion thread or email K6AQ@winlink.org with feedback. 👍
 

<hr /><em>This discussion was created from the release <a href='https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220403'>20220703</a>.</em>

**Comment by kd4vrf** [comment]
I was able to test this out tonight, the duplicates are gone for me in this build. I will run this version this week and let you know if I find anything else.

**Reply by tmiw** [reply]
Sounds good!

**Comment by kd4vrf** [comment]
I've been seeing this off and on since the first build but I haven't mentioned it because I was hoping to find a trigger or pattern to it. At this point I think I have all of the info I'm going to get so I thought I'd share what I have so far.

I have the "Return to this dialog after connection:" option selected on the Pactor Connect window. Under normal usage it closes while it's connecting and after the Pactor session is complete it sends my call and opens the Pactor connect window.

https://user-images.githubusercontent.com/45251519/178075022-2bf2b5a6-48ad-4ced-9e14-5a0c08e49565.mov

You can see in the lower left hand corner of the Paclink window "Pactor FEC", that I believe is sending my call at the end of the transmission. I've never actually listened to it but that's what it looks like on the scope when I transmit, either way it occurs at the end of every normal Pactor connection.

I grabbed four screen recordings of it failing today, here is one of them. If you want to see the logs and screen recordings of the others let me know and I will upload them.


https://user-images.githubusercontent.com/45251519/178075840-f5fc1f90-d93d-4f7d-a8d3-43fc5191fa3e.mov


[Pactor Hang 4 - Logs.zip](https://github.com/ARSFI/paclink-open-source/files/9075230/Pactor.Hang.4.-.Logs.zip)

You can see in this example that I wasn't able to connect to the gateway (I have seen this behavior on both sessions that didn't connect and on successful connections), after the number of attempts is exceeded you see "Disconnected fm N4SER @ 2022/07/08 17:31:51" and "*** Closing Pactor - N4SER at 2022/07/08 17:31:51" but we don't see the "Pactor FEC" in the lower left hand of the Paclink window and the callsign transmission isn't sent. At this point Paclink is still responsive but the Pactor connect dialog box doesn't reappear.

In this example I waited about a minute and a half and the Pactor dialog box doesn't reappear. I have waited longer, sometimes several hours because I walked away and when I came back the Pactor connect dialog still hadn't reappeared.

If you click connect and open another window you immediately see the "Pactor FEC" in the lower left hand corner of the Paclink window and the radio immediately transmits the call sign. After that the Pactor connect dialog box appears and you are able to attempt another connection to an RMS gateway.

Sometimes I'll go days without seeing this behavior. Some days, like today, I'll see it multiple times. I haven't been able to determine a trigger or any pattern to the behavior.

Let me know if you have any questions or if you want to see the logs and recordings of the other times I saw it today.

**Reply by tmiw** [reply]
Sorry for the delay; looks like it was my turn to get sick. (I'm about 95% better now, fortunately.)

Anyway, I was looking at the code and I think I see how the dialog's supposed to be getting recreated. I have a new release at https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220719 with a test change that should hopefully help a bit.

**Comment by kd4vrf** [comment]
No worries, glad to hear you're almost back to 100%.

I got the latest build installed today. I will test it this week and let you know how it goes.
